caffe2/serialize/read_adapter_interface.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+namespace at {
+
+// this is the interface for the (file/stream/memory) reader in
+// PyTorchStreamReader. with this interface, we can extend the support
+// besides standard istream
+class ReadAdapterInterface {
+ public:
+  virtual size_t size() = 0;
```

Comment:
Is there a reason this isn't const? My first intuition would be to make both, `size` and `read` const since I wouldn't expect them to change state.